src/server/custom-functions/metadata.parser.ts

```diff
@@ -57,15 +57,23 @@ function traverseAST(sourceFile: ts.SourceFile): ICFFunctionMetadata[] {
           if (isCF) {
             const jsDocParamInfo = getJSDocParams(func);
 
-            const parameters = func.parameters.map((p: ts.ParameterDeclaration) => {
-              const name = (p.name as ts.Identifier).text;
+            const [lastParameter] = func.parameters.slice(-1);
+            const isStreamingFunction = checkLastParameterForStreaming(lastParameter);
 
-              return {
-                name,
-                ...(jsDocParamInfo[name] ? { description: jsDocParamInfo[name] } : {}),
-                ...getDimAndTypeHelper(p.type),
-              };
-            });
+            const parameters = func.parameters
+              .map((p: ts.ParameterDeclaration, i: number) => {
+                if (isStreamingFunction && i === func.parameters.length - 1) {
```

Comment:
This seems inelegant.  What about just doing a slice of func.parameters if if it a streaming function, skipping the last one?  That would be rather more functional, and cleaner..